solidity/test/FullyBackedECDSAKeepFactoryTest.js

```diff
@@ -0,0 +1,1633 @@
+
+      // Precision (pool weight divisor) is 10^18
+      // (20 * 10^18 * 10) / 10^18                 =  200
+      // ((20 * 10^18 * 10) + (10^18 - 1)) / 10^18 =~ 200.99
+      // Ethereum uint256 division performs implicit floor:
+      // The weight dit not change: 200 == floor(200.99)
+      assert.isTrue(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("returns false if the operator stake dropped below minimum", async () => {
+      await keepFactory.registerMemberCandidate(application, {
+        from: members[0],
+      })
+
+      await setUnbondedValue(members[0], minimumBondableValue.subn(1))
+
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application)
+      )
+    })
+
+    it("reverts if the operator is not registered for the application", async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
+    })
+  })
+
+  describe("updateOperatorStatus", async () => {
+    let minimumBondableValue
+
+    before(async () => {
+      await initializeNewFactory()
+      await initializeMemberCandidates()
+      await registerMemberCandidates()
+
+      const pool = await FullyBackedSortitionPool.at(signerPool)
+      minimumBondableValue = await pool.getMinimumBondableValue()
+    })
+
+    it("revers if operator is up to date", async () => {
+      await expectRevert(
+        keepFactory.updateOperatorStatus(members[0], application),
+        "Operator already up to date"
+      )
+    })
+
+    it("removes operator if bonding value has changed below minimum", async () => {
+      currentValue = await bonding.unbondedValue(members[0])
+
+      const valueToWithdraw = currentValue.sub(minimumBondableValue).addn(1)
+
+      bonding.withdraw(valueToWithdraw, members[0], {from: members[0]})
+      assert.isFalse(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after bonding value change"
+      )
+
+      await keepFactory.updateOperatorStatus(members[0], application)
+
+      await expectRevert(
+        keepFactory.isOperatorUpToDate(members[0], application),
+        "Operator not registered for the application"
+      )
+    })
+
+    it("updates operator if bonding value has changed above minimum", async () => {
+      bonding.deposit(members[0], {value: new BN(1)})
+      assert.isTrue(
+        await keepFactory.isOperatorUpToDate(members[0], application),
+        "unexpected status of the operator after bonding value change"
+      )
+
+      await expectRevert(
+        keepFactory.updateOperatorStatus(members[0], application),
+        "Operator already up to date"
+      )
+    })
```

Comment:
I think we should have another test that would increase this value by more ETH, then assert operator is no longer up to date, update the operator, and assert the operator is up to date again.

The name of this test suggests we are testing the update operation completes successfully but what the test implementation really does is checking that the update is not needed and if executed, it fails.

Reply:
It's already covered pretty well in sortition pool tests https://github.com/keep-network/sortition-pools/blob/678a7b7d10274329fe77b36fe59a2f34e7fc0f65/test/fullyBackedSortitionPoolTest.js#L324-L359

Do we want to duplicate these tests in the factory tests?

Reply:
I think it's fair to have such a test here, especially that we have one for `BondedECDSAKeepFactory`, please see `updates operator if stake has changed above minimum`.

We don't need to assert weight and all sortition-pool specific properties, but `isOperatorUpToDate`/`updateOperatorStatus` are factory's functions and we should cover this behavior with a unit test.

Reply:
It's sortition-pool based logic that we're testing. But, OK I'll duplicate the tests.

Reply:
If we are worried about duplicating tests of sortition pool - and that's fine we are worried about it - we should remove the current test as the fact we add 1 wei to the bond and this update is ignored by the pool is the pool's logic.

We are missing the very basic scenario here when bond has changed and update needs to be performed.

Reply:
The tests we have currently here were added just to test integration with sortition pool. Any test would be fine to cover that. I'll enhance them to cover more cases.

Reply:
A good practice is to test all paths for the component under the test, without going to deep into the test of its dependencies. In this context, we are testing the factory and we don't care what's under `updateOperatorStatus`/`isOperatorUpToDate`. Maybe those functions hit the factory directly and maybe they implement some local caching because hitting the factory every time is expensive. Also, the factory defines the concept of the minimum bond. So I do agree that testing things like no update because the change is insigificant (below pool weight divisor) is not something we should do in tests for the factory, all paths regarding update and minimum stake should be covered.